packages/frontend/app/routes/peers.$peerId.tsx

```diff
@@ -47,10 +53,25 @@ export async function loader({ params }: LoaderArgs) {
 export default function ViewPeerPage() {
   const { peer } = useLoaderData<typeof loader>()
   const response = useActionData<typeof action>()
+  const [formData, setFormData] = useState<FormData>()
+  const submit = useSubmit()
   const { state } = useNavigation()
+  const dialogRef = useRef<ConfirmationDialogRef>(null)
 
   const isSubmitting = state === 'submitting'
 
+  const submitHandler = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
```

Comment:
not having `event.preventDefault()` really destroys everything doesn't it 😆

Reply:
Indeed, removing `event.preventDefault()` would have significant consequences. The form data is processed through the `submit` function, which is obtained from the `useSubmit` hook provided by Remix. When the user clicks the `Delete peer` button, we capture the necessary data to be sent to the server, but we don't immediately submit the form. The actual submission, which is handled within the `onConfirm` function, takes place only when the user clicks the "Delete this peer" button within the dialog.